src/graph/client/graphClient.ts

```diff
@@ -355,29 +355,48 @@ export class GraphClient {
       this.clearGraph();
 
       // Set up nodes and links for the force simulation
-      var nodes: ForceNode[] = vertices
-        .slice(0, maxNodes) // Limit # of nodes
+      var allNodes: ForceNode[] = vertices
         .map(v => <ForceNode>{ vertex: v });
-      var links: ForceLink[] = [];
 
+      // Create map of nodes by ID, and remove duplicates
       var nodesById = new Map<string, ForceNode>();
-      nodes.forEach(n => nodesById.set(n.vertex.id, n));
+      var uniqueNodes: ForceNode[] = [];
+      allNodes.forEach(n => {
```

Comment:
Why don't you just use `vertices` directly?

Reply:
Nodes are a wrapper around vertices